spec/requests/autocomplete_spec.rb

```diff
@@ -11,48 +11,73 @@
       'User-Agent' => 'Stanford Self-Deposit (Happy Heron)'
     }
   end
+  let(:expected_params) do
+    result = ''
+    {
+      query: query,
+      queryIndex: Settings.autocomplete_lookup.query_index,
+      queryReturn: Settings.autocomplete_lookup.query_return,
+      rows: Settings.autocomplete_lookup.num_records,
+      callback: Settings.autocomplete_lookup.callback_function
+    }.each_pair { |k, v| result += "#{k}=#{v}&" }
+    result.chop
+  end
 
   context 'when lookup is successful' do
     let(:status) { 200 }
+    let(:query) { 'mcrae' }
+    let(:suggestions_from_lookup) do
+      [
+        "Bro-C'hall",
+        'Brothers and sisters',
+        'Sisters and brothers',
+        'Television broadcasting',
+        'Brothers',
+        'Brooks',
+        'Bronco busters',
+        'Broncobusters',
+        'Information brokers',
+        'New York (State)--New York--Brooklyn'
+      ]
+    end
     let(:lookup_resp_body) do
       <<~JSON
-        [
-          {
-            "uri": "http://id.worldcat.org/fast/170115",
```

Comment:
I believe we want to record the URI too. Does this still permit that?

Reply:
The best we can do is get idroot back from the request https://www.oclc.org/developer/develop/web-services/fast-api/assign-fast.en.html

We can get "idroot" back from autosuggest, and then manipulate it, but it's not always available (maybe only for subjects)  https://fast.oclc.org/searchfast/fastsuggest?callback=&query=mcrae&queryIndex=suggestall&queryReturn=suggestall,idroot&rows=10

![image](https://user-images.githubusercontent.com/96775/121741474-707b0800-cab3-11eb-9747-0ffc93f7c62d.png)

From their own examples, there seems to be a "suggest" param, but it's not documented:

`suggest=autosuggest` from their first example on their doc page doesn't work, and `suggest=autoSubject` does work but has different results from leaving it out.

Reply:
@ndushay would the Linked Data API have what we need? https://www.oclc.org/developer/develop/web-services/fast-api/linked-data.en.html

Reply:
@arcadia Do you have ideas about what we ought to do when there is no identifier for the topic in FAST?

Reply:
@arcadiafalcone ^^

Reply:
The ones without identifiers are alternate labels and have identifiers under the main label -- for example, "McRae, Col." is an alternate label for "McCrae, John, 1872-1918," which does have an identifier: http://id.worldcat.org/fast/170115. So there may need to be a second lookup to get the main label and identifier. Though that may be weird for the user, as the term will be different from what they select. I suspect that ideally we would only present the main terms to the user to reduce confusion, but I don't know if that's possible.